exporters/otlp/recordable_test.cc

```diff
@@ -0,0 +1,73 @@
+#include "exporters/otlp/recordable.h"
+
+#include <gtest/gtest.h>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace otlp
+{
+TEST(Recordable, SetIds)
+{
+  Recordable rec;
+
+  const uint8_t trace_id_buff[trace::TraceId::kSize]      = {0, 0, 0, 0, 0, 0, 0, 0,
+                                                        0, 0, 0, 0, 0, 0, 0, 1};
+  const uint8_t span_id_buff[trace::SpanId::kSize]        = {0, 0, 0, 0, 0, 0, 0, 2};
+  const uint8_t parent_span_id_buff[trace::SpanId::kSize] = {0, 0, 0, 0, 0, 0, 0, 3};
+
+  const trace::TraceId trace_id(
+      nostd::span<const uint8_t, trace::TraceId::kSize>(trace_id_buff, trace::TraceId::kSize));
+  const trace::SpanId span_id(
+      nostd::span<const uint8_t, trace::SpanId::kSize>(span_id_buff, trace::SpanId::kSize));
+  const trace::SpanId parent_span_id(
+      nostd::span<const uint8_t, trace::SpanId::kSize>(parent_span_id_buff, trace::SpanId::kSize));
+
+  rec.SetIds(trace_id, span_id, parent_span_id);
+
+  EXPECT_EQ(rec.span().trace_id(),
+            std::string(reinterpret_cast<const char *>(trace_id_buff), trace::TraceId::kSize));
+  EXPECT_EQ(rec.span().span_id(),
+            std::string(reinterpret_cast<const char *>(span_id_buff), trace::SpanId::kSize));
+  EXPECT_EQ(rec.span().parent_span_id(),
+            std::string(reinterpret_cast<const char *>(parent_span_id_buff), trace::SpanId::kSize));
+}
+
+TEST(Recordable, SetName)
+{
+  Recordable rec;
+  nostd::string_view name = "TestSpan";
+  rec.SetName(name);
+  ASSERT_EQ(rec.span().name(), name);
+}
+
+TEST(Recordable, SetStartTime)
+{
+  Recordable rec;
+  std::chrono::system_clock::time_point start_time = std::chrono::system_clock::now();
+  core::SystemTimestamp start_timestamp(start_time);
+
+  uint64_t unix_start =
+      std::chrono::duration_cast<std::chrono::nanoseconds>(start_time.time_since_epoch()).count();
+
+  rec.SetStartTime(start_timestamp);
+  ASSERT_EQ(rec.span().start_time_unixnano(), unix_start);
```

Comment:
ASSERT_EQ -> EXPECT_EQ (here and in other tests)

Reply:
Would this be for consistency within the test file? I read about ASSERT vs. EXPECT [here](https://testing.googleblog.com/2008/07/tott-expect-vs-assert.html) and it seems like the only difference is that ASSERT fails right away, where EXPECT continues and fails later on. For the SetIds test, it makes sense to use EXPECT since we don't want the comparison for a certain ID type to block the comparisons of the rest. However, since here we're doing a single comparison, I'm curious what the benefit of using EXPECT would be?

Reply:
Actually, I may be misunderstanding. Does ASSERT terminate the whole test suite or just the specific test case? What I said above applies if it only terminates the specific test case.

Reply:
(Changed everything to EXPECT but I'm still curious about this question)

Reply:
EXPECT allows the current test to continue. ASSERT aborts it. 
It's preferred to use ASSERT when a later part of the test depends on it to be true.